.github/workflows/ci-cd.yml

```diff
@@ -324,18 +344,25 @@ jobs:
         | jq --raw-output .install[].download_info.url
         | tee -a "${GITHUB_OUTPUT}"
       shell: bash
+    - name: Ensure clean for debug
+      if: >-
+        ${{ matrix.debug == 'Y' }}
+      run: >-
+        make clean
+      shell: bash
     - name: >-
         Self-install (from ${{
-          endsWith(matrix.pyver, '-dev')
+          (endsWith(matrix.pyver, '-dev') || matrix.debug == 'Y')
           && 'source'
```

Comment:
I think we can just treat source as debug. Since these aren't compiled. Compiling them in non-debug mode is already exercised in the manylinux jobs. So we can just stick debug into the env vars here and the wheel variants would go through the `-O3` code paths. To achieve this, I'd maybe remove the wheels post download conditionally and the lookup step would only find sdists in such jobs.

Reply:
Left the sdist in because it is needed to hash the requirements file.  It could be refactored more later in a future PR if desired.